test/unit/Query.js

```diff
@@ -800,6 +800,14 @@ describe("Query", () => {
 					});
 				});
 
+				it("Should not throw error if query contains hash key and additional filter expression", () => {
+					queryPromiseResolver = () => ({"Items": []});
+					Model = dynamoose.model("Cat", new dynamoose.Schema({"id": {"type": Number, "hashKey": true}, "name": String}));
+
+					return expect(callType.func(Model.query("id").eq(111).filter("name").eq("Charlie").exec).bind(
+						Model.query("id").eq(111).filter("name").eq("Charlie"))()).to.not.be.rejectedWith("Index can't be found for query.");
```

Comment:
I think maybe we should change this to be a truthy value as opposed to just verifying that it's not rejected with that specific error. Probably checking to ensure the result is an empty array. That way if it returns a different error this test won't pass.

Reply:
Agreed, we should check the output and the throw check will be implied.